docs/markdown/snippets/feature_new.md

```diff
@@ -0,0 +1,32 @@
+## Feature detection based on meson_version in project()
+
+Meson will now print a `WARNING:` message during configuration if you use
+a function or a keyword argument that was added in a meson version that's newer
+than the version specified inside `project()`. For example:
+
+```meson
+project('featurenew', meson_version: '>=0.43')
+
+cdata = configuration_data()
+cdata.set('FOO', 'bar')
+message(cdata.get_unquoted('FOO'))
+```
+
+This will output:
+
+```
+The Meson build system
+Version: 0.47.0.dev1
+Source dir: C:\path\to\srctree
+Build dir: C:\path\to\buildtree
+Build type: native build
+Project name: featurenew
+Project version: undefined
+Build machine cpu family: x86_64
+Build machine cpu: x86_64
+WARNING: Project targetting '>=0.43' but tried to use feature introduced in '0.44.0': get_unquoted
```

Comment:
I think it would be clearer just to show the significant part of the output.

Reply:
I added the whole output to make it clear that the warnings will be emitted in-line as the interpreter passes through the build files, and not all at once in the end. I chose a build file without any compilers defined so that the output would be short(er).